drivers/sdmmc/sdmmc_sdhc.c

```diff
@@ -743,16 +748,20 @@ static bool _wait_for_event(sdhc_dev_t *sdhc_dev,
 #if defined(CPU_SAMD5X) || defined(CPU_SAME5X)
     pm_block(SAM0_PM_IDLE);
 #endif
-    mutex_lock(&sdhc_dev->irq_wait);
+    bool timeout = ztimer_mutex_lock_timeout(_ZTIMER_CLOCK, &sdhc_dev->irq_wait,
+                                             SDHC_IRQ_TIMEOUT_MS) < 0;
 #if defined(CPU_SAMD5X) || defined(CPU_SAME5X)
     pm_unblock(SAM0_PM_IDLE);
 #endif
 
     sdhc->NISIER.reg &= ~event;
     sdhc->EISIER.reg &= ~error_mask;
 
-    if (sdhc_dev->error & error_mask) {
+    if (timeout || (sdhc_dev->error & error_mask)) {
         if (IS_USED(ENABLE_DEBUG)) {
+            if (timeout) {
+                DEBUG("[sdmmc] IRQ wait timeout\n");
+            }
             DEBUG("[sdmmc] SDHC error: EISTR=%04x, ", sdhc_dev->error);
             switch (reset) {
             case SDHC_SRR_SWRSTCMD:
```

Comment:
???
```suggestion
    if (timeout || (sdhc_dev->error & error_mask)) {
        if (IS_USED(ENABLE_DEBUG)) {
            if (timeout) {
                DEBUG("[sdmmc] IRQ wait timeout\n");
            }
            else {
                DEBUG("[sdmmc] SDHC error: EISTR=%04x, ", sdhc_dev->error);
                switch (reset) {
                case SDHC_SRR_SWRSTCMD:
                ...
```
Either a timeout happend or SDHC error happened, right? If so, the DEBUG message should for SDHC error shouldn't be generated in case of timeout error.